gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/HelpBlock.java

```diff
@@ -20,17 +20,63 @@
  * #L%
  */
 
-import com.google.gwt.dom.client.Document;
 import org.gwtbootstrap3.client.ui.base.AbstractTextWidget;
+import org.gwtbootstrap3.client.ui.constants.ElementTags;
+import org.gwtbootstrap3.client.ui.constants.IconType;
 import org.gwtbootstrap3.client.ui.constants.Styles;
 
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.dom.client.Style.Unit;
+
 /**
  * @author Joshua Godi
+ * @author Steven Jardine
  */
 public class HelpBlock extends AbstractTextWidget {
 
+    private IconType iconType = null;
+
     public HelpBlock() {
         super(Document.get().createSpanElement());
         setStyleName(Styles.HELP_BLOCK);
     }
+
+    /**
+     * Sets the icon type.
+     *
+     * @param type the new icon type
+     */
+    public void setIconType(IconType type) {
+        iconType = type;
+    }
+
+    /**
+     * Gets the icon type.
+     *
+     * @return the icon type
+     */
+    public IconType getIconType() {
+        return iconType;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void setText(String text) {
+        super.setText(text);
+        if (iconType != null && text != null && !text.equals("")) {
+            Element e = Document.get().createElement(ElementTags.I);
+            e.addClassName(Styles.FONT_AWESOME_BASE);
+            e.addClassName(iconType.getCssName());
+            e.getStyle().setPaddingRight(5, Unit.PX);
+            getElement().insertFirst(e);
```

Comment:
Yes.  The setText method clears uses Element.setInnerText() which clears the span element.  When the element is cleared we have to re-insert the <i class="fa ...." /> at the beginning of the span in order for it to display.

I though about making the HelpBlock a ComplexWidget and having different elements but I feel this is actually cleaner and accomplishes the task.